assets/pages/digital-subscription-checkout/components/checkoutForm.jsx

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import { countries } from 'helpers/internationalisation/country';
+import { compareString } from 'helpers/utilities';
+
+import { Input } from 'components/forms/standardFields/input';
+import { Select } from 'components/forms/standardFields/select';
+import { withLabel } from 'components/forms/formHOCs/withLabel';
+import { asControlled } from 'components/forms/formHOCs/asControlled';
+
+import { type State, actions } from '../digitalSubscriptionCheckoutReducer';
+
+
+// ----- Form Fields ----- //
+
+function Input1(props: { id: string, label: string, value: string, setValue: string => void }) {
```

Comment:
is `Name1` a convention we use more broadly?

Reply:
I believe @regiskuckaertz used it in places while working on the contributions checkout. It's meant to signify "Input transformed", and I don't think JavaScript allows `'` in variable names. Alternatives would be `InputAsControlledWithLabel`, `NewInput` or `Input_`.

As a side note, I was hoping this intermediate component wouldn't be necessary and that we'd be able to do:

```jsx
const FirstName = compose(
  connect(
    (state: State) => ({ value: state.page.firstName }),
    { setValue: actions.setFirstName },
  ),
  asControlled,
  withLabel,
)(Input);
```

all in one. However, the prop types seem to get lost somewhere in the call to `connect`, so this intermediate function is needed mainly to provide Flow with the type definition.

Reply:
> However, the prop types seem to get lost somewhere in the call to connect

Sad 😕You can probably add type annotations to solve that, though I don't know that it's clearer.

Reply:
Not a huge fan of `Input1` either. `InputField`?

Reply:
@regiskuckaertz haha I literally saw this naming convention in the `Doobie` source code yesterday with `Update0`.